### PR TITLE
feat: Filter Side Bar Enhancements

### LINKED ISSUE #3098: Fix and improve event filters / browse events search
The page https://eventyay.com/explore offers different ways to search for events. Please simplify the search options as follows:
* Add options show "All Categories" and "All Event Types"
* provide a tick box for filters
* move date filters to first position
* add filter option for Call for Speakers "Open", "Closed" and "Not Available"
* add filter option "Free" or "Paid"

Screenshot from 2019-06-10 15-20-13

### LINKED ISSUE #3140: Side Filter Bar Enhancements
Subissue of #3098 

- Addition of All Event Type and Categories. 
- Repositioning of Filters

### PR DESCRIPTION
**Pull request by kushthedude**
Fixes #3140

#### Changes proposed in this pull request:

- Changing Location Of Date Filters
- Addition of `All Categories`
- Addition of `All Types`

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added necessary documentation (if appropriate)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: app/components/explore/side-bar.js [hunk 1]
```diff
@@ -102,6 +112,13 @@ export default Component.extend({
     onDateChange() {
       this.send('selectDateFilter', 'custom_dates');
     },
+    clearFilterCategory() {
+      this.set('category', null);
+      this.set('sub_category', null);
```
Comment (Anupam-dagar) [inline app/components/explore/side-bar.js]: shouldn't we have separate option to clear `sub category` and `category`?

Comment (kushthedude) [reply app/components/explore/side-bar.js]: @Anupam-dagar We are displaying both `SubCategory` and `Category` in one single filter option and I have provided an option for `All Category`, Now providing a way for `All Sub Category` and `All Category` would be tedious

**Comment by kushthedude** [comment]
@shreyanshdwivedi @uds5501 Review 1

File: app/components/explore/side-bar.js [hunk 2]
```diff
@@ -15,6 +15,16 @@ export default Component.extend({
     return !(this.category || this.sub_category || this.event_type || this.startDate || this.endDate || this.location !== null);
   }),
 
+  showAllCategories: computed('category', 'sub_category', function() {
+    return !(this.category || this.sub_category !== null);
+
+  }),
+  showAllTypes: computed('event_type', function() {
+    return !(this.event_type !== null);
```
Comment (niranjan94) [inline app/components/explore/side-bar.js]: Couldn't this be `!this.event_type` ?

File: app/components/explore/side-bar.js [hunk 3]
```diff
@@ -15,6 +15,16 @@ export default Component.extend({
     return !(this.category || this.sub_category || this.event_type || this.startDate || this.endDate || this.location !== null);
   }),
 
+  showAllCategories: computed('category', 'sub_category', function() {
+    return !(this.category || this.sub_category !== null);
```
Comment (niranjan94) [inline app/components/explore/side-bar.js]: This can be `!this.category || !this. sub_category`

File: app/components/explore/side-bar.js [hunk 4]
```diff
@@ -102,6 +112,13 @@ export default Component.extend({
     onDateChange() {
       this.send('selectDateFilter', 'custom_dates');
     },
+    clearFilterCategory() {
+      this.set('category', null);
```
Comment (niranjan94) [inline app/components/explore/side-bar.js]: Use `setProperties` while setting multiple properties

**Comment by kushthedude** [comment]
@niranjan94 Please Reveiw

**Comment by mariobehling** [comment]
Please provide a screenshot.

**Comment by kushthedude** [comment]
@mariobehling 
image

**Comment by kushthedude** [comment]
@shreyanshdwivedi @uds5501 @mrsaicharan1 Please Review

**Comment by kushthedude** [comment]
@niranjan94 @ritikamotwani Review Please

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Please REview

**Comment by kushthedude** [comment]
@niranjan94 Made the changes Please Review

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review Please

File: app/components/explore/side-bar.js [hunk 5]
```diff
@@ -107,8 +117,19 @@ export default Component.extend({
...
+
+    },
+    clearFilterTypes() {
+      this.set('event_type', null);
+    },
 
     clearFilters() {
+
       this.set('startDate', null);
       this.set('endDate', null);
```
Comment (niranjan94) [inline app/components/explore/side-bar.js]: Use `setProperties`

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review Please

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review Please

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review Please

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review Please

**Comment by kushthedude** [comment]
@niranjan94 Please REview

**Comment by kushthedude** [comment]
@CosmicCoder96 Review

**Pull request merged**